File: Laerdal.McuMgr.Bindings.MacCatalystAndIos.Native/McuMgrBindingsiOS/McuMgrBindingsiOS/McuMgrLogLevelHelpers.swift
import Foundation
import iOSMcuManagerLibrary

internal class McuMgrLogLevelHelpers {
    internal static func translateLogLevel(_ minimumNativeLogLevelNumeric: Int) -> McuMgrLogLevel {
        switch minimumNativeLogLevelNumeric {
        case 0:
            return .debug
        case 1:
            return .verbose
        case 2:
            return .info
        case 3:
            return .warning
        case 4:
            return .error
        default:
            return .error
        }
    }
}

File: Laerdal.McuMgr.Bindings.Android.Native/mcumgr-laerdal-wrapper/src/main/java/no/laerdal/mcumgr_laerdal_wrapper/McuMgrLogLevelHelpers.java
package no.laerdal.mcumgr_laerdal_wrapper;

import io.runtime.mcumgr.McuMgrErrorCode;
import io.runtime.mcumgr.exception.McuMgrErrorException;
import io.runtime.mcumgr.response.HasReturnCode;

final class McuMgrLogLevelHelpers
{
    public static EAndroidLoggingLevel translateLogLevel(final int minimumNativeLogLevelNumeric)
    {
        switch (minimumNativeLogLevelNumeric)
        {
            case 0:
                return EAndroidLoggingLevel.Debug;
            case 1:
                return EAndroidLoggingLevel.Verbose;
            case 2:
                return EAndroidLoggingLevel.Info;
            case 3:
                return EAndroidLoggingLevel.Warning;
            case 4:
                return EAndroidLoggingLevel.Error;
            default:
                return EAndroidLoggingLevel.Error;
        }
    }
}
